Build/Build.cs

```diff
@@ -258,11 +275,38 @@ class Build : NukeBuild
         });
 
     Target SpellCheck => _ => _
+        .OnlyWhenDynamic(() => HasDocumentationChanges)
         .Executes(() =>
         {
             Node($"{YarnCli} install --silent", workingDirectory: RootDirectory);
             Node($"{YarnCli} --silent run cspell --no-summary", workingDirectory: RootDirectory,
                 customLogger: (_, msg) => Error(msg));
         });
+
+    bool HasDocumentationChanges =>
+        Changes.Any(x => x.StartsWith("docs"));
+
+    bool HasSourceChanges =>
+        Changes.Any(x => !x.StartsWith("docs"));
+
+    string[] Changes
+    {
+        get
+        {
+            using var repo = new Repository(GitRepository.LocalDirectory);
+
+            string baseRef = BaseRef ?? "develop";
+
+            Tree targetBranch = repo.Branches[baseRef].Tip.Tree;
+            Tree workingDir = repo.Branches[repo.Head.FriendlyName].Tip.Tree;
+
+            return repo.Diff
```

Comment:
One more question. How does this behave during a local build?

Reply:
I don't know if I get you correctly: 
If you have _uncommitted_ changes to the docs, it will ignore the spell check. Otherwise it will run the spell check before the build/test pipeline.

Reply:
In [e90fe4d](https://github.com/fluentassertions/fluentassertions/pull/2089/commits/e90fe4d4c36e2f2d5f29598bfc3660c415f5f0bb) I committed 2 whitespace changes (will remove this commit afterwards)

Reply:
To make sure that this is intentional.

If I:
* check out this PR
* run `git clean -fdx`
* run `./build.ps1`

Then both `SpellCheck` and `Push` are executed.

Reply:
Locally, running `build.ps1` spell check should execute but `Push` doesn't. Why `Push` is executed? There is no tag

Reply:
Sorry, meant that both targets of `Main()` are executed.
`Push` _itself_ is properly skipped as always.

Reply:
Ahh.. yes this is intentional.. I treat them as two individual pipelines, because I had no clue where I should add the spell check semantically

Reply:
And: depending an other target on the documentation spell check.. is a bit odd :D